### DISCUSSION TITLE
Redirect path

### DISCUSSION THREAD

**Discussion by peterhijma** [pocketbase/pocketbase#4168] [Q&A]
I have the following situation:

```
pb_public/
    index.html        (1)  # Simple HTML file
    app/
        index.html    (2)  # React app
        ...
```

The app is a React app, using React Router. The routes are (for simplicity) `/app` and `app/login`.

The following happens:
* When I go to `https://...com/` => I end up at (1), this is good.
* When I go to `https://...com/app` => I end up at (2), this is good. React router does the rest from here.
* **The problem** When I go to `https://...com/app/login` directly, it sends me to (1).

Two questions:
* Shouldn't a non-existing path from the pocketbase point-of-view return a 404?
* How would I get this to work 🤔

**Comment by ganigeorgiev** [comment]
> Shouldn't a non-existing path from the pocketbase point-of-view return a 404?

You can disable the index fallback with `--indexFallback=0` flag when starting the executable (_use `--help` to explore the available options_).

> How would I get this to work

I don't understand what doesn't work.

**Reply by peterhijma** [reply]
Thanks for your quick response. 🙏 

Locally, when using `npm run dev`, I can directly go to `/app/login` and the login page renders. Using pocketbase and a build version of the app inside `pb_public`, pocketbase doesn't recognize `app/login` (of course) and sends me to `/` (or actually, not in the URL, but it shows the template of `/`). My question is: what's the best way to tackle this "problem"?

But maybe this isn't really a pocketbase question 🤔 Sorry in that case

**Reply by ganigeorgiev** [reply]
I'm still not entirely sure that I understand what do you mean.

It also depends in what mode your router is operating - memory, clean urls, hash, etc.
Based on the `app/login` path I assume that you currently use clean urls, right?
If that's the case then what you'll need is a middleware/route that will accept any `app/*` path and will forward it to `app/index.html` (this is the same to what we do with the index fallback).

Alternatively, you can also switch to hash `#/*` based mode (similar to most SPAs, eg. like Gmail) and keep your frontend application router fully static and independent from the server.

**Reply by peterhijma** [reply]
I indeed use clean urls. Feeling not too smart 😢 , but I have no clue how to forward `app/*` to `app/index.html` using the javascript version of hooks. (I did check the docs...)

Also, `app/*` would be a bit too much, because there is a `assets/*.js` folder inside `pb_public/app` as well 😄

**Reply by ganigeorgiev** [reply]
There are several ways to do it but I think the easiest one would be to use the `$apis.staticDirectoryHandler` helper, something like (_I haven't tested it_):

```js
routerAdd("GET", "/app/*", $apis.staticDirectoryHandler("./pb_public/app", true))
```

The above should forward only the missing paths so `app/assets/*` should continue to work.

But if your application is a SPA, it may be better to also switch to a [hash based mode](https://reactrouter.com/en/main/router-components/hash-router). I've checked the React Router documentation and they don't seem to recommend it but there is no explanation why (see also this [stackoverflow thread](https://stackoverflow.com/questions/74199357/why-hashrouter-in-react-router-v6-is-not-recommended)).

**Reply by peterhijma** [reply]
Thanks, it worked perfectly fine! 👍 

I'll also look into the hash based routing, thanks for the tip!
